### PR TITLE
Add Slurm-GCP v6 based solution for provisioning a3-highgpu-8g compute nodes

### PR DESCRIPTION
**Pull request by tpdownes**
This PR is split into 3 commits:

1. Move the a3-highgpu-8g solution from Slurm-GCP v5 to a legacy subdirectory (following pattern for previous v5->v6 migrations documented in examples/README.md)
2. Add prominent notes to the legacy README.md:
- that it is intended for existing clusters and that it should not be used to provision new clusters
- that the v5 solution is tested on v1.37.0 nightly and users should adopt that version of the Toolkit to maintain their clusters
3. Add the new v6 solution based on the existing integration test

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cluster Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-1-image.yaml [hunk 1]
```diff
@@ -92,7 +94,7 @@ deployment_groups:
           set -e -o pipefail
           ansible-galaxy role install googlecloudplatform.google_cloud_ops_agents
           ansible-pull \
-              -U https://github.com/GoogleCloudPlatform/slurm-gcp -C 5.12.0 \
+              -U https://github.com/GoogleCloudPlatform/slurm-gcp -C 6.5.13 \
```
Comment (mr0re1) [inline examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-1-image.yaml:97]: `6.6.0`

File: tools/cloud-build/daily-tests/tests/ml-a3-highgpu-slurm-cluster.yml [hunk 2]
```diff
@@ -41,6 +41,4 @@ cli_deployment_vars:
   server_ip_homefs: "{{ nfs_ip }}"
   remote_mount_homefs: "{{ remote_mount_homefs }}"
   slurm_cluster_name: "{{ slurm_cluster_name }}"
-  disable_login_public_ips: "false"
```
Comment (harshthakkar01) [inline tools/cloud-build/daily-tests/tests/ml-a3-highgpu-slurm-cluster.yml:44]: why we remove this ?

Comment (tpdownes) [reply tools/cloud-build/daily-tests/tests/ml-a3-highgpu-slurm-cluster.yml:44]: The parameters in latest v6 modules are called `enable_login_public_ips` and they are already set to `true`.

File: examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-0-base.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        subnet_private_access: true
+        description: primary subnetwork in $(vars.deployment_name)-sysnet
+    outputs:
+    - network_name
+    - subnetwork_name
+
+  - id: homefs
+    source: modules/file-system/filestore
+    use:
+    - sysnet
```
Comment (harshthakkar01) [inline examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-0-base.yaml:54]: nit: use: [sysnet]

File: examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-2-cluster.yaml [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+    settings:
+      node_count_static: 0
+      node_count_dynamic_max: 4
+      machine_type: n2-standard-2
+      enable_placement: false
+
+  - id: debug_partition
+    source: community/modules/compute/schedmd-slurm-gcp-v6-partition
+    use:
+    - debug_nodeset
```
Comment (harshthakkar01) [inline examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-2-cluster.yaml:103]: nit: use: [debug_nodeset]

File: examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-2-cluster.yaml [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+      node_conf:
+        CoresPerSocket: 52
+        ThreadsPerCore: 2
+      on_host_maintenance: TERMINATE
+      bandwidth_tier: gvnic_enabled
+
+  - id: a3_partition
+    source: community/modules/compute/schedmd-slurm-gcp-v6-partition
+    use:
+    - a3_nodeset
```
Comment (harshthakkar01) [inline examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-2-cluster.yaml:191]: nit: use: [a3_nodeset]

File: examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-2-cluster.yaml [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+
+vars:
+  project_id:  ## Set GCP Project ID Here ##
+  deployment_name: slurm-a3-cluster
+  region: customer-region
+  zone: customer-zone
+  server_ip_homefs: 0.0.0.0 ## MUST set to IP address of Filestore instance from base deployment!
+  remote_mount_homefs: /nfsshare
+  local_mount_homefs: /home
+  zones: []
```
Comment (harshthakkar01) [inline examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-2-cluster.yaml:31]: @mr0re1  will this need to be updated based on #2864 ?

**Review by harshthakkar01 — APPROVED** [review]

File: examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-2-cluster.yaml (hunk 6 above)
Comment (mr0re1) [reply examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-2-cluster.yaml:31]: No, that's good.

**Review by mr0re1 — APPROVED** [review]

File: examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-2-cluster.yaml (hunk 4 above)
Comment (tpdownes) [reply examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-2-cluster.yaml:103]: ack

File: examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-2-cluster.yaml (hunk 5 above)
Comment (tpdownes) [reply examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-2-cluster.yaml:191]: ack

File: examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-1-image.yaml (hunk 1 above)
Comment (tpdownes) [reply examples/machine-learning/a3-highgpu-8g/ml-slurm-a3-1-image.yaml:97]: The tag wasn't available at the time I performed manual testing. The integration test still has known gaps so I prefer to document this as a task to be prioritized soon.

**Pull request merged**